Bottom: d13446196c6c40f87be5fcc2d3f7b8286f57516b
Top:    f5cff464f970eaf8e73e72ff740358f265dd3cd2
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-04-01 13:39:23 +0100

Refresh of add-cache-for-noholes-diced-ve

---

diff --git a/src/polygon1.c b/src/polygon1.c
index 3ad5f22..54cda80 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2313,12 +2313,12 @@ poly_Create (void)
 void
 poly_FreeContours (PLINE **pline)
 {
-  PLINE **pl;
+  PLINE *pl;
 
-  while (*(pl = pline) != NULL)
+  while ((pl = *pline) != NULL)
     {
-      *pline = (*pl)->next;
-      poly_DelContour (pl);
+      *pline = pl->next;
+      poly_DelContour (&pl);
     }
 }
